### ISSUE TITLE
Applying an Image to another Image

### ISSUE THREAD

**Issue by dmyyy** [bevyengine/bevy#16822]
## What problem does this solve or what need does it fill?

I'm personally using this for visualizing a procedural generation pipeline (drawing tiles onto a larger tilemap)

## What solution would you like?

```
pub trait ImageApplyExt {
    fn apply(&mut self, root: UVec2, apply: &Image);
}

impl ImageApplyExt for Image {
    fn apply(&mut self, root: UVec2, apply: &Image) {
        for y in 0..apply.height() {
            for x in 0..apply.width() {
                if let Ok(new_color) = apply.get_color_at(x, y) {
                    if let Err(TextureAccessError::OutOfBounds {..} ) = self.set_color_at(root.x + x, root.y + y, new_color) {
                        // past bounds of base image - nowhere to write to
                        break
                    }
                }
            }
        }
    }
}
```

## What alternative(s) have you considered?

Let users implement this themselves.

**Comment by alice-i-cecile** [comment]
Can you explain exactly what you mean by applying here? Blending? Overwriting? Something else?

**Comment by dmyyy** [comment]
> Can you explain exactly what you mean by applying here? Blending? Overwriting? Something else?

Overwriting (although blending could be added too) - you have a base image and want to overwrite some part of it with another image (like a collage, or a new layer in photoshop).

**Comment by BenjaminBrienen** [comment]
Applying one image to another seems like pretty useful functionality, but if we want to support blending modes, that's a big enough problem to be a whole crate of its own.

**Comment by alice-i-cecile** [comment]
Agreed: we should keep the scope small for now and just do overwriting :)
